geoh5py/objects/surveys/electromagnetics/airborne_tem.py

```diff
@@ -0,0 +1,388 @@
+        "Seconds (s)",
+        "Milliseconds (ms)",
+        "Microseconds (us)",
+        "Nanoseconds (ns)",
+    ]
+    __INPUT_TYPE = ["Rx", "Tx", "Tx and Rx"]
+
+    def __init__(
+        self,
+        object_type: ObjectType,
+        **kwargs,
+    ):
+
+        super().__init__(
+            object_type,
+            **kwargs,
+        )
+
+    def copy(self, parent=None, copy_children: bool = True):
+        """
+        Function to copy a survey to a different parent entity.
+
+        :param parent: Target parent to copy the entity under. Copied to current
+            :obj:`~geoh5py.shared.entity.Entity.parent` if None.
+        :param copy_children: Create copies of all children entities along with it.
+
+        :return entity: Registered Entity to the workspace.
+        """
+        if parent is None:
+            parent = self.parent
+
+        omit_list = ["_metadata", "_receivers", "_transmitters"]
+        new_entity = parent.workspace.copy_to_parent(
+            self, parent, copy_children=copy_children, omit_list=omit_list
+        )
+
+        if "Receivers" in type(self).__name__:
+            new_transmitters = parent.workspace.copy_to_parent(
+                self.transmitters,
+                parent,
+                copy_children=copy_children,
+                omit_list=omit_list,
+            )
+            new_entity.transmitters = new_transmitters
+        else:
+            new_receivers = parent.workspace.copy_to_parent(
+                self.receivers,
+                parent,
+                copy_children=copy_children,
+                omit_list=omit_list,
+            )
+            new_entity.receivers = new_receivers
+
+        parent.workspace.finalize()
+
+        return new_entity
+
+    @property
+    def default_input_types(self) -> list[str]:
+        """Input types. Must be one of 'Rx', 'Tx', 'Tx and Rx'."""
+        return self.__INPUT_TYPE
+
+    @property
+    def default_metadata(self) -> dict:
+        """
+        Default dictionary of metadata for AirborneTEM entities.
+        """
+        return self.__METADATA
+
+    @property
+    def default_units(self) -> list[str]:
+        """Accepted time units. Must be one of "Seconds (s)",
+        "Milliseconds (ms)", "Microseconds (us)" or "Nanoseconds (ns)"
+        """
+        return self.__UNITS
+
+    @property
+    def inline_offset(self):
+        """Inline offset between receiver and transmitter."""
+        if "Inline offset value" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Inline offset value"]
+        if "Inline offset property" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Inline offset property"]
+        return None
+
+    @inline_offset.setter
+    def inline_offset(self, value: float | uuid.UUID):
+        if isinstance(value, float):
+            self.edit_metadata(
+                {"Inline offset value": value, "Inline offset property": None}
```

Comment:
**suggestion:** define constant static variables for re-used literal strings
Each the pair of getter and setter uses the same string as the key to access the metadata.
Instead of copy-pasting the literal string, having these strings defined as constants would be safer, and easier for maintenance (no typo in copy/paste, static analysis ensure the variable exists, easy to change the value at once if it gets renamed in some future version)

Reply:
Implemented, to be reviewed